crates/requirements-txt/src/lib.rs

```diff
@@ -320,24 +321,25 @@ pub struct RequirementsTxt {
 impl RequirementsTxt {
     /// See module level documentation
     #[instrument(skip_all, fields(requirements_txt = requirements_txt.as_ref().as_os_str().to_str()))]
-    pub fn parse(
+    pub async fn parse(
         requirements_txt: impl AsRef<Path>,
```

Comment:
On the question of using `Path`: Part of me feels like we should be using `VerbatimUrl` for this (used elsewhere in this crate). `VerbatimUrl` can represent both URLs and paths, but it preserves the user-provided representation, which we leverage to ensure that we _don't_ expand environment variables (like secrets) when we write results back out to `requirements.txt`. I think we would need a `join` method on it, which would perhaps be somewhat awkward... But would you mind giving it a try, and seeing if it fits? Alternatively, we can punt it to another PR -- I would be open to merging without it.

Reply:
I'd like to give it a try in a separate PR!